src/pyhf/tensor/numpy_backend.py

```diff
@@ -311,8 +305,8 @@ def exp(self, tensor_in: Tensor[T]) -> ArrayLike:
     def percentile(
         self,
         tensor_in: Tensor[T],
-        q: Tensor[T],
-        axis: None | Shape = None,
+        q: float | NDArray[np.floating[T]] | NDArray[np.integer[T]],
+        axis: int | Sequence[int] | None = None,
```

Comment:
@kratsg Are these revisions viable in your mind? Or are they too specific to be be a nuisance later?

Reply:
This is fine, we can type narrow for a specific function. Although maybe using `np.number` is too loose in most places since that allows for complex, and we don't typically allow that for statistical fits...